src/app/DataVault/DataVaultComponent.tsx

```diff
@@ -23,16 +23,22 @@ const DataVaultComponent: React.FC<DataVaultComponentProps> = ({
   addDeclarativeDetail, declarativeDetails, credentials, deleteValue, swapValue, downloadBackup, getKeyContent
 }) => {
   const context = useContext(Web3ProviderContext)
+  const dvClient = context.dvClient
 
-  const handleAdd = (key: string, content: string) =>
-    context.dvClient && addDeclarativeDetail(context.dvClient, key, content)
-  const handleDelete = (key: string, id: string) =>
-    context.dvClient && deleteValue(context.dvClient, key, id)
-  const handleSwap = (key: string, content: string, id: string) =>
-    context.dvClient && swapValue(context.dvClient, key, content, id)
-  const handleDownload = () => context.dvClient && downloadBackup(context.dvClient)
-  const handleGetKeyContent = (key: string) =>
-    context.dvClient && getKeyContent(context.dvClient, key)
+  if (!dvClient) {
+    return (
+      <div className="content data-vault">
+        <h2>Could not connect to the DataVault</h2>
+        <p>Your wallet may be asking for permission.</p>
```

Comment:
Should we add a _retry_ button? Maybe user clicked on cancel or some other strange behaviour. What do you think? We can do it in another PR anyway

Reply:
Yes, it would be a good idea to have a retry button for the DV. I added ticket #58 for this.